ext/sockets/sockets.stub.php

```diff
@@ -2156,3 +2161,21 @@ function socket_wsaprotocol_info_import(string $info_id): Socket|false {}
 
 function socket_wsaprotocol_info_release(string $info_id): bool {}
 #endif
+
+#ifdef AF_PACKET
+final class SocketEthernetInfo
+{
+    /** @readonly **/
+    public Socket $socket;
+    /** @readonly **/
+    public int $ethprotocol;
+    /** @readonly **/
+    public string $macsrc;
+    /** @readonly **/
+    public string $macdst;
+    /** @readonly **/
+    public string $rawpayload;
+    /** @readonly **/
+    public ?object $payload;
```

Comment:
I suggested this, but I think that `rawpayload` is not the right name after all. The `payload` field represents the contents of the eth packet, but the `rawpayload` field represents the eth packet itself. `rawpacket` would be a better name than `rawpayload`.